File: 파이썬/2.검색/1764.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/1764
'''

def binarySearch(arr, length, target) :
    low = 0
    high = length - 1
    while low <=  high :
        mid = (low + high) // 2
        if target == arr[mid] :
            return 1
        elif target < arr[mid] :
            high = mid - 1
        else :
            low = mid + 1
    return 0
 
a, b = map(int, input().split())
a1 = []
b1 = []

for i in range(a) :
    m = input()
    a1.append(m)
    
for i in range(b) :
    m = input()
    b1.append(m)

a1.sort() 

ans = []
for i in range(b) :
    res = binarySearch(a1, a, b1[i])
    if res == 1 :
        ans.append(b1[i])
        
ans.sort()

print(len(ans))
for i in ans :
    print(i)

File: C++/27. 이진 검색 알고리즘 - 초급/1764.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/1764
*/

/*
 * Author : JC
 * https://www.acmicpc.net/problem/1764
*/

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n, m;
string A[500000], B[500000];

int binarySearch(int len, string target){
    int low = 0;
    int high = len-1;
    int mid;
    while(low <= high){
        mid = (low + high) / 2;
        if(A[mid] == target){
            return mid;
        }else if(A[mid] < target){
            low = mid+1;
        }else{
            high = mid-1;
        }
    }
    return -1;
}

int main(){
    
    cin >> n >> m;
    for(int i=0; i<n; i++){
        cin >> A[i];
    }
    for(int i=0; i<m; i++){
        cin >> B[i];
    }
    sort(A, A+n);
    vector<string> v;
    for(int i=0; i<m; i++){
        if(binarySearch(n, B[i]) != -1){
            v.push_back(B[i]);
        }
    }

    sort(v.begin(), v.end());
    cout << v.size() << "\n";
    for(int i=0; i<v.size(); i++){
        cout << v[i] << "\n";
    }
    
    return 0;
}